src/main/java/org/eclipse/uprotocol/rpc/URpcListener.java

```diff
@@ -0,0 +1,36 @@
+package org.eclipse.uprotocol.rpc;
+
+import java.util.concurrent.CompletableFuture;
+import org.eclipse.uprotocol.v1.*;
+
+/**
+ * uService (servers) implement this to receive requests messages from clients. <br>
+ * The service must implement the {@link #onReceive(UUri, UPayload, UAttributes, CompletableFuture)} method to handle
+ * the request and then complete the future passed to the method that triggers the uLink library to
+ * send (over the transport) the response.
+ */
+public interface URpcListener {
```

Comment:
I dont like methods that are void since they are difficult to test
I am not sure why a uE needs to know so much detail
the listener part is an implementation detail and should not be a part of the API like this.
IMHO, there are better ways.
If we find this useful, I am not going to complain, but I think there are better ways.

Reply:
It is very rear to see listeners with returned values in callbacks. Why do we need it? What notifier should do with it?

Reply:
What I don't like is to have to callbacks, it makes harder to use such interfaces as lambda when you want to use not-default callback.

Reply:
@mishap4 fixed this